internal/controller/account_resolver.go

```diff
@@ -29,6 +29,16 @@ type AuthGwResp struct {
 	Identity string `json:"x-rh-identity"`
 }
 
+type ErrorResponse struct {
```

Comment:
Can you rename this to `authGwErrorResponse`?  Its a bit wordy but it communicates its use.

In golang, starting a variable with a capital letter exports the variable out of the package.  In this case, we do not need to expose the Gateway response structures outside of this package.  We really want to hide them from the callers.  So lets not export the error response and the non-error response (please change AuthGwResp to authGwResp).